train.py

```diff
@@ -124,64 +142,111 @@ def train(model, dataset, data_csv, binary, binary_label, params_file, augmentat
+    # image_column = pd.Series([i['image'].numpy() for i in test_dataset_array])
+    # label_column = pd.Series([i['label'].numpy() for i in test_dataset_array])
+    # test_dataset_df = pd.DataFrame({"image": image_column, "label": label_column})
+
+    # # Exclude the test dataset from the training dataset
+    # train_dataset_df = master_dataset_df[~master_dataset_df.index.isin(test_dataset_df.index)]
+
+    # #Make a dataloader for the train dataset
+    # train_dataset_image = torch.tensor(train_dataset_df['image'].values.astype(np.float32)).to(device)
+    # train_dataset_label = torch.tensor(train_dataset_df['label'].values.astype(np.float32)).to(device)
+    # train_dataset = TensorDataset(train_dataset_image, train_dataset_label)
+    # train_plant_dataloader = DataLoader(train_dataset, batch_size=BATCH_SIZE_TRAIN, shuffle=True, num_workers=0)
+
+    # # Make a dataloader for the test dataset
+    # test_dataset_image = torch.tensor(test_dataset_df['image'].values.astype(np.float32)).to(device)
+    # test_dataset_label = torch.tensor(test_dataset_df['label'].values.astype(np.float32)).to(device)
+    # test_dataset = TensorDataset(test_dataset_image, test_dataset_label)
+    # test_plant_dataloader = DataLoader(test_dataset, batch_size=BATCH_SIZE_TRAIN, shuffle=True, num_workers=0)
+
+    # %%
+
+    model_class = get_model_class(MODEL_NAME, num_of_classes=NUM_CLASSES, num_heads=params[MODEL_NAME]['NUM_HEADS'], dropout=params[model]['DROPOUT']).to(device)
+    parameter_grid = {}
+    parameter_grid["lr"] = LR
+    parameter_grid["weight_decay"] = WEIGHT_DECAY
+
+    if OPTIMIZER == "SGD":
+        parameter_grid['dampening'] = float(params[MODEL_NAME]['DAMPENING'])
+        parameter_grid['momentum'] = float(params[MODEL_NAME]['MOMENTUM'])
+        optimizer = optim.SGD(model_class.parameters(), **parameter_grid)
+=======
+    
     if model == 'bag_of_words':
         model_class, y_true, y_pred, test_accuracy, test_loss, other_json = train_bow(master_dataset.df, test_size, NUM_CLASSES, params, save, binary_label)
         train_accuracy = None
         train_loss = None
+>>>>>>> cb0716d121897bb2463d55ac10218bd0c717783f
     else:
-        # hyperparameters:
-        # TODO: the final form of hyperparameters.yaml
         N_EPOCHS = int(params[model]['N_EPOCHS'])
         BATCH_SIZE_TRAIN = int(params[model]['BATCH_SIZE_TRAIN'])
         BATCH_SIZE_TEST = int(params[model]['BATCH_SIZE_TEST'])
         OPTIMIZER = params[model]['OPTIMIZER']
         LR = float(params[model]['LR'])
         WEIGHT_DECAY = float(params[model]['WEIGHT_DECAY'])
-
+        train_dataset, test_dataset = torch.utils.data.random_split(dataset=master_dataset,
+                                        lengths=[train_size + val_size, test_size],
+                                        generator=torch.Generator().manual_seed(42))
         train_dataset, test_dataset = torch.utils.data.random_split(master_dataset, [train_size, test_size])
```

Comment:
```suggestion
```

Reply:
Overrides the previous line where we use the manual seed from hyperparameter search.